### ISSUE TITLE
Resource density in editor on Android failing

### ISSUE THREAD

**Issue by hackedpassword** [yairm210/Unciv#12181]
### Before creating

- [X] This is NOT a gameplay feature from Civ VI, BNW, or outside - see Roadmap
- [X] This is NOT a gameplay feature from Vanilla Civ V or from G&K - If so, it should be a comment in https://github.com/yairm210/Unciv/issues/4697

### Problem Description

Building a custom map. Done mapping, now I want to auto generate resources vs hand place them. Problem is, there's only one setting for resource generation: firehose.

![Screenshot_20240903-035650_Unciv](https://github.com/user-attachments/assets/be2c6043-9294-4964-8a9f-4b41184c7dd5)
![Screenshot_20240903_100915](https://github.com/user-attachments/assets/c6c0d9b8-b909-4f2c-b193-037a34a4e0d2)

### Desired Solution

Showing 'resource richness' under `New Map -> Advanced Settings` even though it has no effect on partial generation. But imo, it should.

### Alternative Approaches

'create ancient ruins' should be controlled by 'resource richness' or another slider.  

### Additional Context

Basically partial map steps look to advanced settings for variance control.

**Comment by yairm210** [comment]
When I change the resource richness and then activate "spread resources" I see that it *did* affect the generation

**Comment by hackedpassword** [comment]
Hmm, I'll look further. Why would it work for you but not me... I'll also try a comparison of android vs pc, see if that has any impact. Thanks for looking, will check back later.

**Comment by hackedpassword** [comment]
Indeed it does. The issue apparently is that on Android, advanced settings aren't respected. Works on the PC.

Android Unciv 4.13.4
PC Unciv 4.12.19

Android 12, galaxy s10 

Guess this feature request turned into a bug report :/

**Comment by yairm210** [comment]
That's weird as hell!
I'll check it out
